Sources/Public/CalendarView.swift

```diff
@@ -734,11 +739,14 @@ public final class CalendarView: UIView {
       isAnimatedUpdatePass: isAnimatedUpdatePass)
     self.anchorLayoutItem = currentVisibleItemsDetails.centermostLayoutItem
 
-    updateVisibleViews(
-      withVisibleItems: currentVisibleItemsDetails.visibleItems,
-      previouslyVisibleItems: visibleItemsDetails?.visibleItems ?? [])
+    // If our first / last layout item hasn't changed, then we haven't scrolled enough to trigger
+    // an update of visible views. This short-circuit greatly improves scroll performance.
+    if currentVisibleItemsDetails.layoutRegion != layoutRegion {
+      updateVisibleViews(withVisibleItems: currentVisibleItemsDetails.visibleItems)
+    }
 
     visibleItemsDetails = currentVisibleItemsDetails
+    layoutRegion = currentVisibleItemsDetails.layoutRegion
```

Comment:
here's the big optimization - when we're scrolling, we only need to `updateVisibleViews(...)` if we've scrolled far enough to change our "layout region" (month headers, days, etc. enter / exit the viewport)